pkgs/development/python-modules/bytewax/default.nix

```
@@ -0,0 +1,73 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, rustPlatform
+, setuptools-rust
+, openssl
+, pkg-config
+, cyrus_sasl
+, protobuf
+, cmake
+, gcc
+, dill
+, multiprocess
+, pytestCheckHook
+, pythonAtLeast
+}:
+
+buildPythonPackage rec {
+  pname = "bytewax";
+  version = "0.15.1";
+  format = "pyproject";
+
+  # Bytewax not supported on 3.11
+  disabled = pythonAtLeast "3.11";
+
+  src = fetchFromGitHub {
+    owner = "bytewax";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-E11IjE5cThVG88ZcpWSgrdjkkHgU05oyT4affEeXCIo=";
+  };
+
+  # Remove docs tests, myst-docutils in nixpkgs is not compatible with package requirements.
+  # Package uses old version.
+  patches = [ ./remove-docs-test.patch ];
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    hash = "sha256-IfVX3k9AsqP84aagCLSwxmutUoEkP8haD+t+VY4V02U=";
+  };
+
+  nativeBuildInputs = [
+  ] ++ (with rustPlatform; [
+    cargoSetupHook
+    maturinBuildHook
+    pkg-config
+    cmake
+    gcc
+  ]);
```

```suggestion
  nativeBuildInputs = [
    cmake
    maturinBuildHook
    pkg-config
    rustPlatform.cargoSetupHook
  ];
```

gcc is part of stdenv

---

updated, but requires rustPlatform.maturinBuildHook